src/graphqlProtocol/subscription.ts

```diff
@@ -0,0 +1,757 @@
+    name: protocolConfiguration.generateSubscriptionNestedInputTypeName(
+      entity,
+      entitySubscription,
+    ),
+    description: `Mutation input type for **\`${typeNamePascalCase}\`** using data uniqueness to resolve references`,
+
+    fields: () => {
+      const fields: GraphQLInputFieldConfigMap = {
+        clientSubscriptionId: {
+          type: GraphQLString,
+        },
+      };
+
+      if (entitySubscription.needsInstance) {
+        fields.nodeId = {
+          type: new GraphQLNonNull(GraphQLID),
+        };
+      }
+
+      if (entitySubscriptionInstanceUniquenessInputType) {
+        fields[typeName] = {
+          type: new GraphQLNonNull(
+            entitySubscriptionInstanceUniquenessInputType,
+          ),
+        };
+      }
+
+      return fields;
+    },
+  });
+
+  return entitySubscriptionInputType;
+};
+
+export const generateSubscriptionOutput = (
+  entity,
+  typeName,
+  type,
+  entitySubscription,
+) => {
+  const protocolConfiguration = ProtocolGraphQL.getProtocolConfiguration() as ProtocolGraphQLConfiguration;
+
+  const typeNamePascalCase = entity.graphql.typeNamePascalCase;
+
+  const entitySubscriptionOutputType = new GraphQLObjectType({
+    name: protocolConfiguration.generateSubscriptionOutputTypeName(
+      entity,
+      entitySubscription,
+    ),
+    description: `Subscription output type for **\`${typeNamePascalCase}\`**`,
+
+    fields: () => {
+      const fields: GraphQLFieldConfigMap<any, any> = {
+        clientSubscriptionId: {
+          type: GraphQLString,
+        },
+      };
+
+      if (entitySubscription.isTypeDelete) {
+        fields.deleteRowCount = {
+          type: new GraphQLNonNull(GraphQLInt),
+          description: 'Number of deleted rows',
+        };
+
+        const primaryAttribute = entity.getPrimaryAttribute();
+
+        if (primaryAttribute) {
+          const fieldName = primaryAttribute.gqlFieldName;
+          const fieldType = ProtocolGraphQL.convertToProtocolDataType(
+            primaryAttribute.type,
+            entity.name,
+            false,
+          );
+
+          fields[fieldName] = {
+            type: new GraphQLNonNull(fieldType),
+            description: primaryAttribute.description,
+          };
+        }
+      } else {
+        fields[typeName] = {
+          type: new GraphQLNonNull(type),
+        };
+      }
+
+      return fields;
+    },
+  });
+
+  return entitySubscriptionOutputType;
+};
+
+const extractIdFromNodeId = (graphRegistry, sourceEntityName, nodeId) => {
+  let instanceId;
+
+  if (nodeId) {
+    const { type, id } = fromGlobalId(nodeId);
+
+    instanceId = id;
+
+    const entity = graphRegistry.types[type]
+      ? graphRegistry.types[type].entity
+      : null;
+
+    if (!entity || entity.name !== sourceEntityName) {
+      throw new Error('Incompatible nodeId used with this mutation');
+    }
+  }
+
+  return instanceId;
+};
+
+export const generateSubscriptions = graphRegistry => {
```

Comment:
The difference with mutation resides in the field return in the subscription map, as GraphQL specify a Subscription field with type, description, args, subscribe (returning the AsyncIterator), and an optional resolve.